src/scenario/scenario_events_xml.h

```diff
@@ -0,0 +1,638 @@
+#ifndef SCENARIO_EVENTS_XML_H
+#define SCENARIO_EVENTS_XML_H
+
+#include "building/type.h"
+#include "city/message.h"
+#include "game/resource.h"
+#include "game/settings.h"
+#include "scenario/data.h"
+#include "scenario/scenario_event_data.h"
+
+typedef enum {
+    PARAMETER_TYPE_UNDEFINED = -1,
+    PARAMETER_TYPE_NUMBER = 0,
+    PARAMETER_TYPE_TEXT = 1,
+    PARAMETER_TYPE_CHECK = 2,
+    PARAMETER_TYPE_DIFFICULTY = 3,
+    PARAMETER_TYPE_RESOURCE = 4,
+    PARAMETER_TYPE_BOOLEAN = 5,
+    PARAMETER_TYPE_ROUTE = 6,
+    PARAMETER_TYPE_POP_CLASS = 7,
+    PARAMETER_TYPE_BUILDING = 8,
+    PARAMETER_TYPE_BUILDING_COUNTING = 9,
+    PARAMETER_TYPE_ALLOWED_BUILDING = 10,
+    PARAMETER_TYPE_STANDARD_MESSAGE = 11,
+    PARAMETER_TYPE_FUTURE_CITY = 12,
+    PARAMETER_TYPE_MIN_MAX_NUMBER = 13,
+} parameter_type;
+
+typedef struct {
+    const char *name;
+    parameter_type type;
+    int min_limit;
+    int max_limit;
+} xml_data_attribute_t;
+
+typedef struct {
+    condition_types type;
+    xml_data_attribute_t xml_attr;
+    xml_data_attribute_t xml_parm1;
+    xml_data_attribute_t xml_parm2;
+    xml_data_attribute_t xml_parm3;
+    xml_data_attribute_t xml_parm4;
+    xml_data_attribute_t xml_parm5;
+} scenario_condition_data_t;
+
+static scenario_condition_data_t scenario_condition_data[CONDITION_TYPE_MAX] = {
+    [CONDITION_TYPE_TIME_PASSED]     = { .type = CONDITION_TYPE_TIME_PASSED,
+                                        .xml_attr =     { .name = "time",           .type = PARAMETER_TYPE_TEXT },
+                                        .xml_parm1 =    { .name = "check",          .type = PARAMETER_TYPE_CHECK,            .min_limit = 1,         .max_limit = 3 },
+                                        .xml_parm2 =    { .name = "min",            .type = PARAMETER_TYPE_MIN_MAX_NUMBER,           .min_limit = 0,         .max_limit = 1000000000 },
+                                        .xml_parm3 =    { .name = "max",            .type = PARAMETER_TYPE_MIN_MAX_NUMBER,           .min_limit = 0,         .max_limit = 1000000000 },
+                                        .xml_parm4 =    { .name = "",               .type = PARAMETER_TYPE_UNDEFINED },
```

Comment:
If you set `PARAMETER_TYPE_UNDEFINED` to `0` on the `enum` above, you won't need to fill values for unused elements.

All you may have to check is for the `.xml_paramX` against null pointers instead of empty strings, but depending on your code you probably won't even need to change anything at all (I still haven't checked `scenario_events_xml.c` at the time of writing this).

Reply:
I've done the undefined as 0.
Still looking into what the impact of not defaulting to empty string is.

Reply:
Done. I've made the name change as well.